keras_nlp/layers/modeling/rotary_embedding.py

```diff
@@ -16,6 +16,40 @@
 
 
 class RotaryEmbedding(keras.layers.Layer):
+    """Rotary positional encoding layer.
+
+    This layer encodes absolute positional information with rotation
+    matrix. It calculates the rotary encoding with a mix of sine and
+    cosine functions with geometrically increasing wavelengths.
+    Defined and formulated in [RoFormer: Enhanced Transformer with Rotary Position Embedding](https://arxiv.org/abs/2104.09864v4).
+    Takes as input the query and key tensors. The input must have shape
+    [batch_size, num_heads, sequence_length, query_length]. This layer
+    will return new query and key tensors after applying rotational
+    encoding.
+
+    Args:
+        percentage: float. The percentage of attn_head_size over which rotation
+            should be applied. Defaults to 0
+        max_wavelength: int. The maximum angular wavelength of the sine/cosine
+            curves, as described in Attention is All You Need. Defaults to
+            `10000`.
+
+    Examples:
+
+    ```python
+    batch_size = 16
+    num_heads = 8
+    sequence_length = 256
+    query_length = 256
+    query = tf.ones((batch_size, num_heads, sequence_length, query_length))
+    rot_emb_layer = RotaryEmbedding()
+    query_rot = rot_emb_layer(query)
+    ```
+
+    References:
+     - [RoFormer: Enhanced Transformer with Rotary Position Embedding](https://arxiv.org/abs/2104.09864v4)
+    """
+
     def __init__(self, max_wavelength=10000, **kwargs):
```

Comment:
We should probably add two arguments here. `sequence_axis=1`, and `feature_axis=-1`, that users can set as desired.

Reply:
This would be cool. Added.